core/templates/dev/head/domain/collection/CollectionObjectFactory.js

```diff
@@ -203,6 +223,85 @@ oppia.factory('CollectionObjectFactory', [
       return skillList;
     };
 
+    // Gets a new ID for a skill. This should be of the same form as in the
+    // backend.
+    Collection.prototype.getNewSkillId = function() {
+      var newId = 's' + String(this._skillIdCount);
+      this._skillIdCount++;
+      return newId;
+    };
+
+    // Adds a new frontend collection skill domain object to this collection.
+    // This will return true if the skill was successfully added, or false if
+    // the given collection skill has an ID already used by another skill, the
+    // ID is invalid, or another skill already has the same name.
+    // Changes to the provided object will be reflected in this collection.
+    Collection.prototype.addCollectionSkill = function(collectionSkillObject) {
+      if (!this._skills.hasOwnProperty(collectionSkillObject.getId())) {
+        if (Number(collectionSkillObject.getId().slice(1)) >=
```

Comment:
Should we perform this validation here or as part of the collection validation service?